src/models/event.ts

```diff
@@ -404,8 +404,12 @@ export class MatrixEvent extends TypedEventEmitter<MatrixEventEmittedEvents, Mat
         });
 
         this.txnId = event.txn_id;
+        // The localTimestamp is calculated using the age.
+        // Some homeserver don't provide the unsigned field for state events.
+        // The fallback in these cases will be to use the origin_server_ts.
+        // For EDUs, the origin_server_ts also is not defined so we use Date.now().
         const age = this.getAge();
-        this.localTimestamp = age ? Date.now() - age : this.getTs() ?? 0;
+        this.localTimestamp = age ? Date.now() - age : this.getTs() ?? Date.now();
```

Comment:
we should probably do:
```suggestion
        this.localTimestamp = age !== undefined ? Date.now() - age : this.getTs() ?? Date.now();
```
 to guard against the unlikely but possible case that `age` is zero.